### PR TITLE
chore: Add discussion on SDK 2.0

### PR DESCRIPTION
**Pull request by marikaner**
Feel free to add your ideas

### CODE REVIEW

File: knowledge-base/architectural-decisions-records/sdk-2.0.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Get rid of moment
+
+## Other backlog items
+
+## Other ideas
+* What about a generic vdm?
+  * Release vdm using edmx files + generic vdm
+  * maybe generate on install?
+* Is is possible to rename VDM to @sap-cloud-sdk-vdm/xy-service
```
Comment (marikaner) [inline knowledge-base/architectural-decisions-records/sdk-2.0.md:60]:
```suggestion
* Is is possible to rename VDM to @sap-cloud-sdk-vdm/xy-service
* Should we have a shared eslint ruleset for contributions / demo projects?
```

File: knowledge-base/architectural-decisions-records/sdk-2.0.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+* Should we have a shared eslint ruleset for contributions / demo projects?
```
Comment (marikaner) [inline knowledge-base/architectural-decisions-records/sdk-2.0.md:61]:
```suggestion
* Should we have a shared eslint ruleset for contributions / demo projects?
* We can already rename `GetAllRequestBuilderV2` -> `GetAllRequestBuilder`, etc. by exporting files like this:
  * `export {GetAllRequestBuilder as GetAllRequestBuilderV2}`
```

File: knowledge-base/architectural-decisions-records/sdk-2.0.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * `@sap-cloud-sdk/odata-v2`
+ * `@sap-cloud-sdk/odata-v4`
+ * `@sap-cloud-sdk/connectivity`
+ * `@sap-cloud-sdk/http-client`
+If needed maybe also `@sap-cloud-sdk/core` but hopefully not needed.
+* Sort packages into separate directories beforehand (#before)
+  * Mimic packages (without package.json and co) in the core directory
+  * Circular depencies - make sure that there are only imports from the whole module, e. g.
+    In odata-v2 package, if I want to import X from http-client import it from `http-client` not `http-client/X`
+* The connectivity package currently has only the cf scope. If at some point we add other systems (e. g. kyma) this will be added here.
```
Comment (jjtang1985) [inline knowledge-base/architectural-decisions-records/sdk-2.0.md:26]: 👍 
I think for the time being, it makes lot of sense to keep relatively smaller number of packages.
If we make `connectivity-cf`, then `connectivity-gcp`, the dependency tree `@sap-cloud-sdk/http-client` -> `@sap-cloud-sdk/connectivity` shown below will also be more complicated.

File: knowledge-base/architectural-decisions-records/sdk-2.0.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+### Package dependencies (#after, conceptually #before)
+The packages above should rely on one another like this (dependant -> dependency):
+* `@sap-cloud-sdk/odata-v2` -> `@sap-cloud-sdk/odata-common`
+* `@sap-cloud-sdk/odata-v4` -> `@sap-cloud-sdk/odata-common`
+* `@sap-cloud-sdk/odata-common` -> `@sap-cloud-sdk/connectivity`
+* `@sap-cloud-sdk/odata-common` -> `@sap-cloud-sdk/http-client`
+* `@sap-cloud-sdk/http-client` -> `@sap-cloud-sdk/connectivity`
+
+### Other related activities (#after)
+Rename all functionality with V2/V4 suffix, e. g.:
```
Comment (jjtang1985) [inline knowledge-base/architectural-decisions-records/sdk-2.0.md:37]: maybe update a bit, to reflect the latest plan:
- rename now, but export `GetAllRequestBuilderV2`.
- adjust the export later

File: knowledge-base/architectural-decisions-records/sdk-2.0.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+## Other backlog items
+
+## Other ideas
+* What about a generic vdm?
+  * Release vdm using edmx files + generic vdm
+  * maybe generate on install?
+* Is is possible to rename VDM to @sap-cloud-sdk-vdm/xy-service
+* Should we have a shared eslint ruleset for contributions / demo projects?
+* We can already rename `GetAllRequestBuilderV2` -> `GetAllRequestBuilder`, etc. by exporting files like this:
```
Comment (jjtang1985) [inline knowledge-base/architectural-decisions-records/sdk-2.0.md:63]: I think we agreed this, so we can move this up.

**Review by jjtang1985 — APPROVED** [review]

**Comment by FrankEssenberger** [comment]
We need to add a timeline in the next meeting. BTW do we keep this PR open until we are finished discussing? Then I would add a do not merge label?

**Comment by jjtang1985** [comment]
I think we can merge it first, then next time you can just review the delta part.

**Pull request merged**